### PR TITLE
[api] replace api server with api serverV2

### PR DESCRIPTION
**Pull request by Liuhaai**
delayed until #2886 is merged

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -74,33 +74,33 @@ var (
...
-}
-
-// WithNativeElection is the option to return native election data through API.
-func WithNativeElection(committee committee.Committee) Option {
-	return func(cfg *Config) error {
-		cfg.electionCommittee = committee
-		return nil
-	}
-}
+// // BroadcastOutbound sends a broadcast message to the whole network
```
Comment (raullenchai) [inline api/api.go:77]: just remove unused code, don't do this

Comment (Liuhaai) [reply api/api.go]: removed. Once the test has passed, the file `api.go` and `api_test.go` will be deleted in the next PR

File: api/coreservice.go [hunk 2]
```diff
@@ -259,7 +259,7 @@ func (core *coreService) ServerMeta() (packageVersion string, packageCommitID st
 }
 
 // SendAction is the API to send an action to blockchain.
-func (core *coreService) SendAction(ctx context.Context, in *iotextypes.Action, chainID uint32) (string, error) {
+func (core *CoreService) SendAction(ctx context.Context, in *iotextypes.Action, chainID uint32) (string, error) {
 	log.L().Debug("receive send action request")
 	var selp action.SealedEnvelope
 	if err := selp.LoadProto(in); err != nil {
```
Comment (dustinxie) [inline api/coreservice.go:265]: line 270, change `IsKamchatka` to `IsToBeEnabled`

File: chainservice/chainservice.go [hunk 3]
```diff
@@ -631,8 +630,8 @@ func (cs *ChainService) ActionPool() actpool.ActPool {
 }
 
 // APIServer returns the API server
-func (cs *ChainService) APIServer() *api.Server {
-	return cs.api
+func (cs *ChainService) APIServer() *api.CoreService {
+	return cs.api.CoreService()
```
Comment (dustinxie) [inline chainservice/chainservice.go:634]: i think just return `cs.api` here, so no need to change `coreService` to `CoreService`

File: e2etest/local_transfer_test.go [hunk 4]
```diff
@@ -406,7 +406,7 @@ func TestLocalTransfer(t *testing.T) {
 				if err1 != nil {
 					return err1
 				}
-				selp, err = as.GetActionByActionHash(tsfHash)
+				selp, err = as.ActionByActionHash(tsfHash)
```
Comment (dustinxie) [inline e2etest/local_transfer_test.go:409]: these changes are not necessary, see above comment

File: e2etest/rewarding_test.go [hunk 5]
```diff
@@ -532,17 +532,17 @@ func injectClaim(
...
-	api *api.Server,
+	api *api.CoreService,
 	exptUnclaimed map[string]*big.Int,
 	claimedAmount map[string]*big.Int,
 	pendingClaimActions map[hash.Hash256]bool,
 ) bool {
 	updated := false
 	for selpHash, expectedSuccess := range pendingClaimActions {
-		receipt, err := api.GetReceiptByActionHash(selpHash)
+		receipt, err := api.ReceiptByActionHash(selpHash)
```
Comment (dustinxie) [inline e2etest/rewarding_test.go:542]: these changes are not necessary, see above comment

File: tools/minicluster/minicluster.go [hunk 6]
```diff
@@ -205,7 +205,7 @@ func main() {
 		// Wait until the smart contract is successfully deployed
 		var receipt *action.Receipt
 		if err := testutil.WaitUntil(100*time.Millisecond, 60*time.Second, func() (bool, error) {
-			receipt, err = svrs[0].ChainService(uint32(1)).APIServer().GetReceiptByActionHash(eHash)
+			receipt, err = svrs[0].ChainService(uint32(1)).APIServer().ReceiptByActionHash(eHash)
```
Comment (dustinxie) [inline tools/minicluster/minicluster.go:208]: these changes are not necessary, see above comment

File: e2etest/rewarding_test.go [hunk 7]
```diff
@@ -532,17 +532,17 @@ func injectClaim(
 
 func updateExpectationWithPendingClaimList(
 	t *testing.T,
-	api *api.Server,
+	api *api.CoreService,
```
Comment (dustinxie) [inline e2etest/rewarding_test.go:535]: *api.ServerV2

File: api/grpcserver.go [hunk 8]
```diff
@@ -347,3 +349,15 @@ func (svr *GrpcServer) GetLogs(ctx context.Context, in *iotexapi.GetLogsRequest)
 	}
 	return &iotexapi.GetLogsResponse{Logs: ret}, err
 }
+
+// TODO: remove internal call GetActionByActionHash()
+
+// GetActionByActionHash returns action by action hash
+func (svr *GrpcServer) GetActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error) {
```
Comment (dustinxie) [inline api/grpcserver.go:356]: i am thinking move this to `apiServerV2.go`?
so it becomes an internal/utility func of api, not GrpcServer

File: e2etest/local_transfer_test.go [hunk 9]
```diff
@@ -496,7 +496,7 @@ func TestLocalTransfer(t *testing.T) {
 			require.NoError(err, tsfTest.message)
 			tsfHash, err1 := tsf.Hash()
 			require.NoError(err1)
-			_, err = as.GetActionByActionHash(tsfHash)
+			_, err = as.GRPCServer().GetActionByActionHash(tsfHash)
```
Comment (dustinxie) [inline e2etest/local_transfer_test.go:499]: and we can revert these changes

File: e2etest/rewarding_test.go [hunk 10]
```diff
@@ -532,17 +533,19 @@ func injectClaim(
...
-	api *api.Server,
+	api *api.ServerV2,
 	exptUnclaimed map[string]*big.Int,
 	claimedAmount map[string]*big.Int,
 	pendingClaimActions map[hash.Hash256]bool,
 ) bool {
 	updated := false
 	for selpHash, expectedSuccess := range pendingClaimActions {
-		receipt, err := api.GetReceiptByActionHash(selpHash)
+		receipt, err := api.GRPCServer().GetReceiptByAction(context.Background(), &iotexapi.GetReceiptByActionRequest{
```
Comment (dustinxie) [inline e2etest/rewarding_test.go:543]: do the same for `GetReceiptByAction`, move to `apiServerV2.go`

File: e2etest/rewarding_test.go [hunk 11]
```diff
@@ -532,17 +533,19 @@ func injectClaim(
...
+			ActionHash: hex.EncodeToString(selpHash[:]),
+		})
 
 		if err == nil {
-			selp, err := api.GetActionByActionHash(selpHash)
+			selp, err := api.GRPCServer().GetActionByActionHash(selpHash)
```
Comment (dustinxie) [inline e2etest/rewarding_test.go:548]: and we can revert these changes

File: tools/util/injectorutil.go [hunk 12]
```diff
@@ -693,9 +693,12 @@ func CheckPendingActionList(
...
 	pendingActionMap.RangeEvictOnError(func(selphash cache.Key, vi interface{}) error {
 		empty = false
-		receipt, err := cs.APIServer().GetReceiptByActionHash(selphash.(hash.Hash256))
+		sh, _ := selphash.(hash.Hash256)
+		receipt, err := cs.APIServer().GRPCServer().GetReceiptByAction(context.Background(), &iotexapi.GetReceiptByActionRequest{
+			ActionHash: hex.EncodeToString(sh[:]),
+		})
 		if err == nil {
-			selp, err := cs.APIServer().GetActionByActionHash(selphash.(hash.Hash256))
+			selp, err := cs.APIServer().GRPCServer().GetActionByActionHash(selphash.(hash.Hash256))
```
Comment (dustinxie) [inline tools/util/injectorutil.go:701]: and we can revert these changes

File: chainservice/chainservice.go [hunk 13]
```diff
@@ -630,9 +630,15 @@ func (cs *ChainService) ActionPool() actpool.ActPool {
 	return cs.actpool
 }
 
+// APIServer defines the interface of core service of the server
+type APIServer interface {
+	ActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error)
+	ReceiptByAction(h string) (*action.Receipt, string, error)
```
Comment (dustinxie) [inline chainservice/chainservice.go:636]: also use `hash.Hash256`

File: e2etest/rewarding_test.go [hunk 14]
```diff
@@ -532,17 +533,17 @@ func injectClaim(
...
-	api *api.Server,
+	api chainservice.APIServer,
 	exptUnclaimed map[string]*big.Int,
 	claimedAmount map[string]*big.Int,
 	pendingClaimActions map[hash.Hash256]bool,
 ) bool {
 	updated := false
 	for selpHash, expectedSuccess := range pendingClaimActions {
-		receipt, err := api.GetReceiptByActionHash(selpHash)
+		receipt, _, err := api.ReceiptByAction(hex.EncodeToString(selpHash[:]))
```
Comment (dustinxie) [inline e2etest/rewarding_test.go:543]: then no need to call `hex.EncodeToString`

File: tools/minicluster/minicluster.go [hunk 15]
```diff
@@ -205,7 +206,7 @@ func main() {
 		// Wait until the smart contract is successfully deployed
 		var receipt *action.Receipt
 		if err := testutil.WaitUntil(100*time.Millisecond, 60*time.Second, func() (bool, error) {
-			receipt, err = svrs[0].ChainService(uint32(1)).APIServer().GetReceiptByActionHash(eHash)
+			receipt, _, err = svrs[0].ChainService(uint32(1)).APIServer().ReceiptByAction(hex.EncodeToString(eHash[:]))
```
Comment (dustinxie) [inline tools/minicluster/minicluster.go:209]: same here

File: tools/util/injectorutil.go [hunk 16]
```diff
@@ -693,9 +693,10 @@ func CheckPendingActionList(
 
 	pendingActionMap.RangeEvictOnError(func(selphash cache.Key, vi interface{}) error {
 		empty = false
-		receipt, err := cs.APIServer().GetReceiptByActionHash(selphash.(hash.Hash256))
+		sh, _ := selphash.(hash.Hash256)
+		receipt, _, err := cs.APIServer().ReceiptByAction(hex.EncodeToString(sh[:]))
```
Comment (dustinxie) [inline tools/util/injectorutil.go:697]: same here

**Review by CoderZhi — APPROVED** [review]

File: chainservice/chainservice.go [hunk 17]
```diff
@@ -632,13 +632,13 @@ func (cs *ChainService) ActionPool() actpool.ActPool {
 
 // APIServer defines the interface of core service of the server
 type APIServer interface {
-	ActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error)
-	ReceiptByAction(h hash.Hash256) (*action.Receipt, string, error)
```
Comment (dustinxie) [inline chainservice/chainservice.go]: this interface is used internally, IMO no need to change to protobuf data struct

**Review by dustinxie — APPROVED** [review]
